Implements Locust (#37)
hellofresh/kangal: docs/locust/README.md

# Locust

## Table of content
- [How it works](#how-it-works)
- [Configuring Locust resource requirements](#configuring-locust-resource-requirements)
- [Writing tests](#writing-tests)
- [Reporting](#reporting)

> Locust is one of the load generators used in Kangal and uses the official docker image [locustio/locust](https://hub.docker.com/r/locustio/locust).

Reviewer: ```suggestion
Locust is one of the load generators implemented in Kangal. It uses the official docker image [locustio/locust](https://hub.docker.com/r/locustio/locust).
```

# Locust

## Table of content
- [Configuring Locust resource requirements](#configuring-locust-resource-requirements)
- [Writing tests](#writing-tests)
- [Reporting](#reporting)

> Locust is one of the load generators used in Kangal. An easy to use, scriptable and scalable performance testing tool. You define the behaviour of your users in regular Python code, instead of using a clunky UI or domain specific language. This makes Locust infinitely expandable and very developer friendly.

Author: Sure, I will remove, I copied from Locust official repo

> Locust is one of the load generators used in Kangal. An easy to use, scriptable and scalable performance testing tool. You define the behaviour of your users in regular Python code, instead of using a clunky UI or domain specific language. This makes Locust infinitely expandable and very developer friendly.

Author: Done!

Currently Kangal uses the official docker image [locustio/locust](https://hub.docker.com/r/locustio/locust).

Kangal requires a py testfile describing the test.

Check it out [Locust official website](https://locust.io/).

## Configuring Locust resource requirements
By default, Kangal does not specify resource requirements for loadtests run with Locust as a backend.

You can specify resource limits and requests for Locust master and worker containers to ensure that when the load generator runs, it has sufficient resources and will not fail before the service does.

The following environment variables can be specified to configure this parameter:

```
LOCUST_MASTER_CPU_LIMITS
LOCUST_MASTER_CPU_REQUESTS
LOCUST_MASTER_MEMORY_LIMITS
LOCUST_MASTER_MEMORY_REQUESTS
LOCUST_WORKER_CPU_LIMITS
LOCUST_WORKER_CPU_REQUESTS
LOCUST_WORKER_MEMORY_LIMITS
LOCUST_WORKER_MEMORY_REQUESTS
```

> You have to specify this variables on Kangal Controller, read more at [charts/kangal/README.md](/charts/kangal/README.md#kangal-controller-locust-specific).

Reviewer: ```suggestion
You have to specify these variables on Kangal Controller, read more at [charts/kangal/README.md](/charts/kangal/README.md#kangal-controller-locust-specific).
```
